File: src/main/java/com/github/tuk-cps/ExprParser/ExprTreeNot.java
package ExprParser;

import jAADD.BDD;

/**
 * Predefined functions: not
 */
class ExprTreeNot extends ExprTreeFunction {
    ExprTreeNot() { super(BDD.BOOL, "not", new ExprTree("x", BDD.BOOL)); }
    void evalThis() throws ExprError {
        if (param.size() != 1) throw new ExprError("not expects 1 parameter");
        result = getBDDParam(0).negate();
    }
}

File: src/main/kotlin/com/github/tuk-cps/exprParser/ExprTreeNot.kt
package exprParser

import jAADD.BDD.Companion.variable

/**
 * Predefined functions: not
 */
internal class ExprTreeNot : ExprTreeFunction(variable("x"), "not", ExprTree(variable("x"), "x")) {
    @Throws(ExprError::class)
    override fun evalUp() {
        if (param.size != 1) throw ExprError("not expects 1 parameter")
        value = getBDDParam(0).not()
    }
}